### PR TITLE
Fix search

### PR DESCRIPTION
**Pull request by hakubo**

### CODE REVIEW

**Review by rybmat — APPROVED** [review]

**Review by rybmat — APPROVED** [review]

File: app/components/wikia-search.js [hunk 1]
```diff
@@ -265,7 +276,7 @@ export default Component.extend(
 				.catch((reason) => this.get('logger').error('Search suggestions error', reason))
 				.finally(() => {
 					// We have a response, so we're no longer loading the results
-					if (phrase === this.get('phrase')) {
+					if (phrase === this.get('phrase') || !this.get('isDestroyed')) {
```
Comment (latata) [inline app/components/wikia-search.js]: `phrase === this.get('phrase') && !this.get('isDestroyed')`

File: app/components/wikia-search.js [hunk 2]
```diff
@@ -245,7 +256,7 @@ export default Component.extend(
 							 * Also, we don't want to show the suggestion results after a real search
 							 * will be finished, what will happen if search request is still in progress.
 							 */
-							if (!this.get('searchRequestInProgress') && phrase === this.get('phrase')) {
+							if ((!this.get('searchRequestInProgress') && phrase === this.get('phrase'))) {
```
Comment (latata) [inline app/components/wikia-search.js]: why did you add extra brackets?

**Review by latata — COMMENTED** [review]
Looks like we should definitely consider using `ember-concurency` :)

**Review by latata — APPROVED** [review]

**Review by rybmat — APPROVED** [review]
